### DISCUSSION TITLE
How to query a tree structure from self-relation schema?

### DISCUSSION THREAD

**Discussion by Rackar** [prisma/prisma#16061] [Q&A] [answered]
I have a demo schema:

```prisma
model testDepartment {
  id       Int              @id @default(autoincrement())
  name     String
  parent   testDepartment?  @relation("parent", fields: [parentId], references: [id])
  children testDepartment[] @relation("parent")
  parentId Int?
}
```

I use a nest include to query the children:
```ts
let include: any = undefined
    for (let i = 0; i < level; i++) {
      if (i === 0) {
        include = { children: true }
      } else {
        include = { children: { include: include } }
      }
    }
    let data = await prisma.testDepartment.findFirst({
      where: { id },

      include: include,
    })
```

It looks bad. Is there any suggestion?

**Comment by pantharshit00** [comment, marked as answer]
Hello @Rackar,

Currently, the approach you have mentioned would be way of achieving the desired result here. There is an outstanding feature request going on about improving support for tree structure: https://github.com/prisma/prisma/issues/4562#issuecomment-860176071. You can leave a "+1" as a reaction to the first comment to help prioritise it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
